app/assets/javascripts/preferences/panes/account/changePassword/index.tsx

```diff
@@ -0,0 +1,192 @@
+import { useState } from '@node_modules/preact/hooks';
+import ModalDialog, {
+  ModalDialogButtons,
+  ModalDialogDescription,
+  ModalDialogLabel
+} from '@/components/shared/ModalDialog';
+import { Button } from '@/components/Button';
+import { FunctionalComponent } from 'preact';
+import { WebApplication } from '@/ui_models/application';
+import ChangePasswordSuccess from '@/preferences/panes/account/changePassword/ChangePasswordSuccess';
+import ChangePasswordForm from '@/preferences/panes/account/changePassword/ChangePasswordForm';
+import useBeforeUnload from '@/hooks/useBeforeUnload';
+
+enum SubmitButtonTitles {
+  Default = 'Continue',
+  GeneratingKeys = 'Generating Keys...',
+  Finish = 'Finish'
+}
+
+enum Steps {
+  InitialStep,
+  FinishStep
+}
+
+type Props = {
+  onCloseDialog: () => void;
+  application: WebApplication;
+}
+
+const ChangePassword: FunctionalComponent<Props> = ({
```

Comment:
Actually I decomposed my originally created `ChangePassword.tsx` to 3 components. I decided not to use the same "ChangePassword" name for both the folder and its main component, therefore named it `index.tsx`. This is a common approach and this way we also just import `ChangePassword` by just `.../account/changePassword` instead of `.../account/changePassword/ChangePassword`.